gemrb/core/Scriptable/Scriptable.cpp

```diff
@@ -1727,11 +1727,17 @@ void Highlightable::DrawOutline(Point origin) const
 	}
 	origin = outline->BBox.origin - origin;
 
-	if (core->HasFeature(GFFlags::PST_STATE_FLAGS)) {
-		VideoDriver->DrawPolygon(outline.get(), origin, outlineColor, true, BlitFlags::MOD | BlitFlags::HALFTRANS);
-	} else {
-		VideoDriver->DrawPolygon( outline.get(), origin, outlineColor, true, BlitFlags::BLENDED|BlitFlags::HALFTRANS );
-		VideoDriver->DrawPolygon( outline.get(), origin, outlineColor, false );
+	bool highlightOutlineOnly = core->HasFeature(GFFlags::HIGHLIGHT_OUTLINE_ONLY);
+	bool pstStateFlags = core->HasFeature(GFFlags::PST_STATE_FLAGS);
+
+	if (!highlightOutlineOnly) {
+		BlitFlags flag = BlitFlags::HALFTRANS | (pstStateFlags ? BlitFlags::MOD : BlitFlags::BLENDED);
+
+		VideoDriver->DrawPolygon(outline.get(), origin, outlineColor, true, flag);
+	}
+
+	if (highlightOutlineOnly || !pstStateFlags) {
```

Comment:
The pst check is now redundant here, so it can be removed.

Reply:
Why is it redundant exactly? According to the previous code we should not draw the outline if the pst flag is set.

Reply:
Now there's a configuration option available with the same effect. If anyone wanted to change the behaviour, it would unexpectedly have no effect, which is bad.

Reply:
Ah ok, now I understand where you're coming from. Still, as long as we want to have the previous behaviour with the pst flag available alongside the new option, we need such a test:

- HOO=0, PST=0 : (fill with BlitFlags::BLENDED) + (outline)
- HOO=0, PST=1 : (fill with BlitFlags::MOD)
- HOO=1, PST=0 : (outline)
- HOO=1, PST=1 : (outline)

The first two cases represent the previous behaviour for which we need the test. It is the last case which generates a conflict in which option has precedence. But how to resolve? Always draw the outline? Would that be ok for PST?

Reply:
Now, what does the PST_STATE_FLAG actually control in the context here? It controls the BlitFlag. That's actually all it does here. So I think we can resolve the conflict with always drawing the outline, but we also apply the BlitFlags to the outline which is depending on the pst flag.

With this we can seperate the concerns of both flags:
- HOO: only draw outline
- PST: different blending/color

Reply:
Ah, I see, I was just looking at pst, not everyone else.  So it's the highlight check that's redundant in that line comparing to the old behaviour. But this way it does become configurable.